src/Tests/Microsoft.DotNet.ApiCompatibility.Tests/Rules/AttributesMustMatchTests.cs

```diff
@@ -220,41 +220,46 @@ public class First {}
     CompatDifference.CreateWithDefaultMetadata(DiagnosticIds.CannotChangeAttribute, "", DifferenceType.Changed, "T:CompatTests.First:[T:CompatTests.FooAttribute]")
 }
             },
-            // Attributes with internal type arguments
+            // Attributes on internal type arguments
             {
                 @"
 namespace CompatTests
 {
   using System;
+  using CompatTestsSecondNamespace;
 
   [AttributeUsage(AttributeTargets.All, AllowMultiple = true)]
   public class FooAttribute : Attribute {
     public FooAttribute(Type type) {}
-    public Type A;
   }
 
-  internal class Bar {}
-
-  [Foo(typeof(Bar), A = typeof(int))]
+  [Foo(typeof(Bar))]
   public class First {}
 }
+
+namespace CompatTestsSecondNamespace {
+  internal class Bar {}
+}
 ",
                 @"
 namespace CompatTests
 {
   using System;
-
   [AttributeUsage(AttributeTargets.All, AllowMultiple = true)]
   public class FooAttribute : Attribute {
     public FooAttribute(Type type) {}
     public Type A;
   }
 
-  [Foo(typeof(int), A = typeof(int))]
+  internal class Bar {}
+
+  [Foo(typeof(Bar))]
   public class First {}
 }
 ",
-new CompatDifference[] {}
+new CompatDifference[] {
+    CompatDifference.CreateWithDefaultMetadata(DiagnosticIds.CannotChangeAttribute, "", DifferenceType.Changed, "T:CompatTests.First:[T:CompatTests.FooAttribute]"),
```

Comment:
Thank you for adding the test - you may want to also include some comments that explain why it's important to examine internal types.